### PR TITLE
expose actual errors rather than silently swallowing them when loading zkCoordinator config file

### PR DESCRIPTION
**Pull request by suicidejack**
This might be an issue with the other configuration load logic as well but I have not checked yet.  If you agree with what this PR is doing then I can check around and fix the others as well.

### CODE REVIEW

**Comment by serejja** [comment]
Hey @suicidejack, thanks for noticing this. Yes, this looks like an issue indeed and your PR is certainly useful. Looking through the other places quickly I noticed the same thing in ConsumerConfig loading as well so if you'd fix that as well it would be really great.

Thanks!

**Comment by suicidejack** [comment]
@serejja updated the other configuration file load functions.  I also added API documentation on the config files so it's easier to understand what those files are without having to look through the code to find the field names.

File: consumer_config.go [hunk 1]
```diff
@@ -330,6 +330,41 @@ func (c *ConsumerConfig) Validate() error {
...
 }
 
+// ConsumerConfigFromFile is a helper function that loads a consumer's configuration from file.
+// The file accepts the following fields:
+//  group.id
+//  consumer.id
+//  fetch.message.max.bytes
+//  num.consumer.fetchers
+//  rebalance.max.retries
+//  queued.max.message.chunks=
```
Comment (serejja) [inline consumer_config.go]: A minor typo here

File: consumer_config.go [hunk 2]
```diff
@@ -330,6 +330,41 @@ func (c *ConsumerConfig) Validate() error {
...
+//  fetch.batch.size
+//  fetch.batch.timeout
+//  requeue.ask.next.backoff
+//  fetch.max.retries
+//  fetch.topic.metadata.retries
+//  fetch.topic.metadata.backoff
+//  fetch.request.backoff
+//  blue.green.deployment.enabled
+// The configuration file entries should be constructed in key=value syntax. A # symbol at the beginning
+// of a line indicates a comment. Blank lines are ignored.
```
Comment (serejja) [inline consumer_config.go]: Maybe it's also worth mentioning the configuration file must end with a new line as stated here: https://github.com/jimlawless/cfg/blob/master/cfg.go#L67

File: producer_config.go [hunk 3]
```diff
@@ -46,6 +46,22 @@ func DefaultProducerConfig() *ProducerConfig {
...
+//  flush.byte.count
+//  flush.timeout
+//  batch.size
+//  max.message.bytes
+//  max.messages.per.request
+//  acks
+//  retry.backoff
+//  timeout
+// The configuration file entries should be constructed in key=value syntax. A # symbol at the beginning
+// of a line indicates a comment. Blank lines are ignored.
```
Comment (serejja) [inline producer_config.go]: Same here regarding the ending with a new line

File: zk_coordinator.go [hunk 4]
```diff
@@ -976,6 +976,15 @@ func NewZookeeperConfig() *ZookeeperConfig {
...
 
+// ZookeeperConfigFromFile is a helper function that loads zookeeper configuration information from file.
+// The file accepts the following fields:
+//  zookeeper.connect
+//  zookeeper.kafka.Root
+//  zookeeper.connection.timeout
+//  zookeeper.max.request.retries
+//  zookeeper.request.backoff
+// The configuration file entries should be constructed in key=value syntax. A # symbol at the beginning
+// of a line indicates a comment. Blank lines are ignored.
```
Comment (serejja) [inline zk_coordinator.go]: And here

**Comment by serejja** [comment]
@suicidejack so the PR looks really cool now, should probably change the typo thing and MAYBE mention the config file must end with a new line. If you don't feel it's worth mentioning the newline stuff or we can switch to another property file parsing library feel free to comment.

Thanks!

File: consumer_config.go (hunk 1 above)
Comment (serejja) [inline consumer_config.go]: Well not a typo, just an equals sign in the end

File: consumer_config.go (hunk 2 above)
Comment (suicidejack) [inline consumer_config.go]: Good catch!  That wasn't mentioned on the godoc ;)

**Comment by suicidejack** [comment]
@serejja good catches!  I updated the PR based on your comments (as you can tell I like clear explicit API documentation :stuck_out_tongue_winking_eye: ).  

Unrelated to this PR (let me know if you would rather discuss somewhere else): Are there any open tasks or assistance that you need with this library still?  You have the only implementation of a HL consumer in go and we (my work) will be heavily relying on this (under very heavy volume).  We are happy to invest some time into this library if you need the help.  Also, what is the status of this package: https://github.com/stealthly/go-kafka?  Are you planning on implementing a "normal producer" vs the syslog producer at some point (or is this an outstanding task)?

**Comment by joestein** [comment]
Hey @suicidejack do you want to connect with me on linked in https://www.linkedin.com/in/charmalloc and then we can email and talk? I can give you an update of where we are and what is going on with your different projects/repo and if there are tasks that are interesting to you would love the help, for sure!

**Comment by serejja** [comment]
@suicidejack looks awesome now. Thanks for your contribution!

**Pull request merged**